tools.py

```diff
@@ -112,8 +112,8 @@ def git_add_and_pr(self, gem, matlab_output):
 
     def check_dependencies(self):
         # Check Matlab version
-        cmd = sp.check_output(['/usr/local/bin/matlab', '-nodisplay -nosplash -nodesktop -r', '"disp(version); quit"'])
-        m_version = ' '.join(cmd.decode('utf-8').split()[-1:]).replace('(','').replace(')','')
+        cmd = sp.check_output(['/usr/local/bin/matlab', '-nodisplay -nosplash -nodesktop -batch', '"disp(version); quit"'])
```

Comment:
Are you sure `-batch` has an effect here?

Reply:
yes, it does. When running the command in `-r` mode, the total stored output in `cmd` is [here in line 7](https://github.com/SysBioChalmers/ecModels/runs/1937708470?check_suite_focus=true#step:3:7), which corresponds to a concatenation between some kind of MATLAB welcome message + the output of the `disp(version)` command. 

After noticing this, my logic was the following: let's try to get rid of the welcome message and just get the version string, then parse the full `version` string and store it as it is. 

The closer that I got to remove the undesired output part was by running the command in `batch` mode, which returns a much shorter message + the full `version` string. You can see it [here in lines 8-9](https://github.com/SysBioChalmers/ecModels/runs/1939003569?check_suite_focus=true).

Reply:
You're right, I confused this with `batch mode`. [Here](https://www.mathworks.com/help/matlab/ref/matlabwindows.html) is what `-batch` does:

> To start MATLAB non-interactively, use the -batch option. To start MATLAB interactively, use the -r option.

In this case, I think this option should be added to all places where Matlab is invoked.